src/contract.rs

```diff
@@ -1059,6 +1089,12 @@ pub fn execute_update_config(
         }
     }
 
+    if let Some(tos_version) = tos_version.clone() {
```

Comment:
I don't understand the purpose of checking if the config version matches our config for a config update execution.